### PR TITLE
feat(chips): typescript support

### PR DESCRIPTION
**Pull request by moog16**
related #48

### CODE REVIEW

**Comment by appsforartists** [comment]
This is a massive diff that I don't have time to look at right now.  This stood out:

You use `@ts-ignore`.  I don't recognize that rule. (I'm guessing it's some kind of linter.) You should include _why_ you're using it in the comments, so as the codebase evolves, future editors understand if they should remove it, and auditors understand what it does.

File: packages/chips/Chip.tsx [hunk 1]
```diff
@@ -50,10 +89,10 @@ export class Chip extends Component {
...
   }
 
-  init = (el) => {
+  init = (el: HTMLDivElement) => {
     this.chipElement_ = el;
     this.props.initRipple(el);
-  }
+  };
 
   get classes() {
```
[the remark this answers is not in the document]
Comment (moog16) [reply packages/chips/Chip.tsx]: This is an implied string return type. If you hover over it in vscode, you can see it being applied. Also we have `inferredAny: false`. So its built into our linter that this cannot happen.

File: packages/chips/Chip.tsx [hunk 2]
```diff
@@ -19,16 +19,55 @@
...
+  removeIcon?: React.ReactElement<HTMLElement>;
+};
+
+type ChipState = {
+  classList: Set<string>;
+  leadingIconClassList: Set<string>;
+};
+
+export class Chip extends React.Component<ChipProps, ChipState> {
+  chipElement_: HTMLDivElement | null = null;
```
[the remark this answers is not in the document]
Comment (moog16) [reply packages/chips/Chip.tsx]: this is being used in the tests and cannot be private :(

File: packages/chips/Chip.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rlfriedman) [reply packages/chips/Chip.tsx]: Cool!

File: packages/chips/Chip.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (rlfriedman) [reply packages/chips/Chip.tsx]: Ah, ok

**Pull request merged**